keps/sig-node/1287-in-place-update-pod-resources/README.md

```diff
@@ -99,48 +101,89 @@ Other identified non-goals are:
-  denotes the Node resources **allocated** to the Pod and its Containers,
-* Pod.Status.ContainerStatuses[i].Resources (new object, type
+* Pod.Status.ContainerStatuses[i].ResourcesAllocated (new field, type
+  v1.ResourceList) denotes the Node resources **allocated** to the Pod and its
+  Containers,
+* Pod.Status.ContainerStatuses[i].Resources (new field, type
   v1.ResourceRequirements) shows the **actual** resources held by the Pod and
   its Containers.
+* Pod.Status.ContainerStatuses[i].Resize (new field, type map[string]string)
+  explains what is happening for a given resource on a given container.
+
+The new `ResourcesAllocated` field represents in-flight resize operations and
+is driven by state kept in the node checkpoint.  Schedulers should user the
+larger of `Spec.Containers[i].Resources` and
+`Status.ContainerStatuses[i].ResourcesAllocated` when considering available
+space on a node.
 
-A new admission controller named 'PodResourceAllocation' is introduced in order
-to limit access to ResourcesAllocated field such that only Kubelet can update
-this field.
+#### Subresource
 
-Additionally, Kubelet is authorized to update PodSpec, and NodeRestriction
-admission plugin is extended to limit Kubelet's update access only to Pod's
-ResourcesAllocated field for CPU and memory resources.
+For alpha, resource changes will be made by updating the pod spec.  For beta
+(or maybe a followup in alpha), a new subresource, /resize, will be defined.
+This subresource could eventually apply to other resources that carry
+PodTemplates, such as Deployments, ReplicaSets, Jobs, and StatefulSets.  This
+will allow users to grant RBAC access to controllers like VPA without allowing
+full write access to pod specs.
+
+The exact API here is TBD.
 
 #### Container Resize Policy
 
 To provide fine-grained user control, PodSpec.Containers is extended with
-ResizePolicy - a list of named subobjects (new object) that supports 'cpu'
-and 'memory' as names. It supports the following policy values:
-* NoRestart - the default value; resize Container without restarting it,
-* RestartContainer - restart the Container in-place to apply new resource
-  values. (e.g. Java process needs to change its Xmx flag)
-
-By using ResizePolicy, user can mark Containers as safe (or unsafe) for
-in-place resource update. Kubelet uses it to determine the required action.
+ResizePolicy - a list of named subobjects (new object) that supports 'cpu' and
+'memory' as names. It supports the following policy values:
+* RestartNotRequired - default value; try to resize the Container without
+  restarting it, if possible.
+* Restart - the container requires a restart to apply new resource values.
+  (e.g.  Java process needs to change its Xmx flag) By using ResizePolicy, user
+  can mark Containers as safe (or unsafe) for in-place resource update. Kubelet
+  uses it to determine the required action.
+
+Note: `RestartNotRequired` does not guarantee that container won't be
```

Comment:
Could you please list some specific scenario for `RestartNotRequired`, i.e. The runtime have to stop the container.

Reply:
Were you looking for [motivation section](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources#motivation) in the original KEP? This PR is a design update to the original KEP, and I hope the above has the answer you're looking for.